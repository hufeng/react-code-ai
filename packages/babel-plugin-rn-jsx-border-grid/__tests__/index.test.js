const babel = require('babel-core');
const borderGrid = require('../index');

describe('babel-plugin-rn-jsx-border-grid test suite', () => {
  it('test no @showGrid', () => {
    const tpl = `
      const Hello = () => <View>
        <Text>hello world</Text>
      </View>
    `;

    const { code } = babel.transform(tpl, {
      plugins: [borderGrid, 'syntax-jsx']
    });

    expect(code).toMatchSnapshot();
  });

  it('test @showGrid and no style', () => {
    const tpl = `
    /*
    @showGrid
    */
    const Hello = () => <View>
      <Text>hello world</Text>
    </View>
  `;

    const { code } = babel.transform(tpl, {
      plugins: [borderGrid, 'syntax-jsx']
    });

    expect(code).toMatchSnapshot();
  });

  it('test @showGriud and has style', () => {
    const tpl = `
    /*
    @showGrid
    */
    const Hello = () => <View style={{flex:1}}>
      <Text style={{color: 'yellow'}}>hello world</Text>
    </View>
  `;

    const { code } = babel.transform(tpl, {
      plugins: [borderGrid, 'syntax-jsx']
    });

    expect(code).toMatchSnapshot();
  });

  it('test @showGriud and has ref style', () => {
    const tpl = `
    /*
    @showGrid
    */
    const Hello = () => <View style={styles.container}>
      <Text style={[styles.text, hasBackground && {backgroundColor: 'yellow'}]}>
        hello world
      </Text>
    </View>
  `;

    const { code } = babel.transform(tpl, {
      plugins: [borderGrid, 'syntax-jsx']
    });

    expect(code).toMatchSnapshot();
  });
});
