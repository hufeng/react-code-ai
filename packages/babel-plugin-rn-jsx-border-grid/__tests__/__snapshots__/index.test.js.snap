// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-rn-jsx-border-grid test suite test @showGrid and no style 1`] = `
"
/*
@showGrid
*/
const Hello = () => <View style={{
  borderWidth: 1,
  borderColor: \\"#FFD700\\",
  borderStyle: \\"dotted\\"
}}>
      <Text style={{
    borderWidth: 1,
    borderColor: \\"#FFFF00\\",
    borderStyle: \\"dotted\\"
  }}>hello world</Text>
    </View>;"
`;

exports[`babel-plugin-rn-jsx-border-grid test suite test @showGriud and has ref style 1`] = `
"
/*
@showGrid
*/
const Hello = () => <View style={[styles.container, {
  borderWidth: 1,
  borderColor: '#00FF33',
  borderStyle: 'dotted'
}]}>
      <Text style={[styles.text, hasBackground && { backgroundColor: 'yellow' }, {
    borderWidth: 1,
    borderColor: '#FF00FF',
    borderStyle: 'dotted'
  }]}>
        hello world
      </Text>
    </View>;"
`;

exports[`babel-plugin-rn-jsx-border-grid test suite test @showGriud and has style 1`] = `
"
/*
@showGrid
*/
const Hello = () => <View style={[{ flex: 1 }, {
  borderWidth: 1,
  borderColor: '#33FFCC',
  borderStyle: 'dotted'
}]}>
      <Text style={[{ color: 'yellow' }, {
    borderWidth: 1,
    borderColor: '#FF1493',
    borderStyle: 'dotted'
  }]}>hello world</Text>
    </View>;"
`;

exports[`babel-plugin-rn-jsx-border-grid test suite test no @showGrid 1`] = `
"
const Hello = () => <View>
        <Text>hello world</Text>
      </View>;"
`;
